scripts/generate_tilesets.py

```diff
@@ -456,24 +484,19 @@ def main(cfg_file_path):
                     logger.warning(f'{nbr_duplicated_id} tiles were in common to the GT, OTH and FP datasets')
                     logger.warning(f'{initial_nbr_gt_tiles - final_nbr_gt_tiles} GT tiles were removed because of their presence in the FP or OTH dataset.')
 
-                aoi_tiles_gdf = pd.concat([
-                    aoi_tiles_intersecting_gt_labels.head(DEBUG_MODE_LIMIT//2), # a sample of tiles covering GT labels
-                    aoi_tiles_intersecting_fp_labels.head(DEBUG_MODE_LIMIT//4), # a sample of tiles convering FP labels
-                    aoi_tiles_intersecting_oth_labels.head(DEBUG_MODE_LIMIT//4), # a sample of tiles convering OTH labels
-                    aoi_tiles_gdf # the entire tileset, so as to also have tiles covering no label at all (duplicates will be dropped)
-                ])
-
+                if FP_LABELS:
+                    aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_fp_labels)
+                elif OTH_LABELS:
+                    aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_oth_labels)
+                else:
+                    aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_fp_labels, aoi_tiles_intersecting_oth_labels)
```

Comment:
Here, either `FP_LABELS` or `OTH_LABELS` is true since you could enter this code section. It means you will always enter the `if` or the `elif`, but never the `else`. To keep this form, you would need to check that one is `False`, _i.e._ `None`, when the other is `True`, _i.e._ a gdf.

The most elegant solution I can think of is either:
```
if not FP_LABELS:
     aoi_tiles_intersecting_fp_labels = gpd.GeoDataFrame()
elif not OTH_LABELS:
     aoi_tiles_intersecting_oth_labels = gpd.GeoDataFrame()
aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_fp_labels, aoi_tiles_intersecting_oth_labels)
```
or (but I don't know if this would produce the same result):
```
if FP_LABELS:
     aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_fp_labels)
if OTH_LABELS:
     aoi_tiles_gdf = concat_sampled_tiles(DEBUG_MODE_LIMIT, aoi_tiles_gdf, aoi_tiles_intersecting_gt_labels, aoi_tiles_intersecting_oth_labels)
```